### PR TITLE
Add --cpus-per-task to srun commands for Hera and Orion

### LINKED ISSUE #2044: Add --cpus-per-task to APRUN commands for Hera and Orion
### What new functionality do you need?

After update to SLURM version on Orion we need to now set the `--cpus-per-task` flag with the srun command.

Also add flag for srun commands on Hera for consistency even though workaround was implemented by sysadmins there.

Issue #1996 documents a slowdown in the analysis job (`gsi.x`), troubleshooting with Orion helpdesk, and resulting solution tested by @RussTreadon-NOAA . A hotfix is being made to add the flag for the analysis job. Now investigate adding the flag for all job srun commands on Orion.

Comments from helpdesk:
```
Russ, I've been digging into this, and I spotted something this time. You and another user who's been having a 
similar slowdown issue with a completely different package have the same sbatch directive:

#SBATCH --cpus-per-task=10

Per the slurm release page:

"NOTE: Beginning with 22.05, srun will not inherit the --cpus-per-task value requested by salloc or sbatch. It 
must be requested again with the call to srun or set with the SRUN_CPUS_PER_TASK environment variable if 
desired for the task(s). "

I have actually just been testing using

#SBATCH --ntasks=400

since this test is nodes=10, with 40 tasks per node, and it appears to be MUCH faster.

For your submission, you should use the same I believe, --ntasks=400.
```
```
The following change was implemented on NOAA's R&D HPC systems after upgrading
to version 22.05.6.

There has been a change of behavior in Slurm going from 21.08.8-2 to 22.05.6.
In 21, when srun was run from an sbatch-submitted job script, it inherited the
cpus-per-task from the sbatch environment. In 22, srun seems to default to
cpus-per-task=2 instead of inheriting from the sbatch environment. This causes
massive slowdowns in jobs which are expecting the 21.x inheritance behavior.

We are adding the following line to Hera's task prolog:
echo export SRUN_CPUS_PER_TASK=$SLURM_CPUS_PER_TASK

This line sets the contents of the SRUN_CPUS_PER_TASK variable (used by srun)
to whatever the existing contents of the SLURM_CPUS_PER_TASK were, thus
emulating the 21.x behavior.
```

### What are the requirements for the new functionality?

The new flag works on Hera and Orion.

### Acceptance Criteria

Jobs run at same speed (or faster). Evaluate speed.

### Suggest a solution (optional)

Add `--cpus-per-task $THREAD_VAR` to each job in `HERA.env` and `ORION.env`.

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
# Description

This PR adds the `--cpus-per-task` flag to non-CFP `APRUN`/srun commands in the env files for Orion and Hera. This is needed as a result of an upgrade to SLURM on Orion that no longer exported the thread value to the environment. Issue #1996 was a hotfix to add this flag for the analysis job that was suddenly running slowly after the SLURM upgrade. A similar SLURM upgrade was done on Hera but the sysadmins there implemented a workaround so this issue did not appear there.

This PR adds the `--cpus-per-task` flag for both machines...needed on Orion and as a safety measure on Hera.

Also included in this PR is a fix to indentations in the `scripts/exglobal_diag.sh` script that was noticed during work for this PR.

Timing differences were not observed for other jobs after adding this srun flag. Job timings were of a similar variation between control and test runs. Any potential additional speedup was not observed. The test runs also reproduced the control runs with `develop`.

Resolves #2044

# Type of change

- Bug fix (fixes something broken)
- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

- Cycled tests on Hera and Orion against controls
- Coupled forecast-only test on Orion against control

### CODE REVIEW

**Review by aerorahul — APPROVED** [review]
looks good.

**Comment by aerorahul** [comment]
Adding the JEDI developers to review the changes to their specific tasks.

**Review by RussTreadon-NOAA — APPROVED** [review]
@KateFriedman-NOAA reports no change in analysis, forecast, or downstream results & no impact on job run times.

`atmanalrun` and `atmensanlrun` changes look correct.  Did not test to confirm they work as intended.

Approve pending successful completion of CI.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI tests.

**Review by guillaumevernieres — APPROVED** [review]
Thanks @KateFriedman-NOAA .

**Review by jiaruidong2017 — APPROVED** [review]
Looks good to me.

**Review by DavidHuber-NOAA — APPROVED** [review]
Since ESMF handles threading, this doesn't need to be added to the forecast jobs, correct?

**Comment by KateFriedman-NOAA** [comment]
> Since ESMF handles threading, this doesn't need to be added to the forecast jobs, correct?

@DavidHuber-NOAA As I understand it, yes.

File: env/HERA.env [hunk 1]
```diff
@@ -43,7 +43,7 @@ if [[ "${step}" = "prep" ]] || [[ "${step}" = "prepbufr" ]]; then
 
 elif [[ "${step}" = "preplandobs" ]]; then
 
-    export APRUN_CALCFIMS="${launcher} -n 1"
+    export APRUN_CALCFIMS="${launcher} -n 1" # TODO: Does this need threading (--cpus-per-task flag)?
```
Comment (CoryMartin-NOAA) [inline env/HERA.env]: I don't think so, for this or the other land job

Comment (KateFriedman-NOAA) [reply env/HERA.env]: Thanks @CoryMartin-NOAA ! I have removed those comments.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**